### DISCUSSION TITLE
Plots notebook vs .py

### DISCUSSION THREAD

**Discussion by tenFnAN** [posit-dev/positron#5034] [Q&A] [answered]
I don't understand the difference in the quality of plots created in an IPython notebook vs. a pure Python script.

For example:

`sns.pairplot(data=ts_ml[['y', 'y_lag_10', 'y_lag_51', 'y_lag_52', 'y_lag_100', 'y_lag_102']] )
plt.show()`

The plot in the notebook looks as expected: 
![obraz](https://github.com/user-attachments/assets/2d695c83-6f40-4ab9-aba0-10a15c1c3749)
However, when plotting the same in a .py script: 
![obraz](https://github.com/user-attachments/assets/942f8782-9aca-49cd-96c6-369c94a245be)

Are there any plotting options I should inspect more closely?

**Comment by jthomasmock** [comment, marked as answer]
Hi @tenFnAN -- thanks for the question and for trying out Positron.

For seaborn/matplotlib, the matplotlib “Size and Shape” setting is pretty much identical to VS Code notebook and the "default" settings for size. Let me know if the below is helpful

Here's the setting for sizing policies:
![image](https://github.com/user-attachments/assets/cdaaf025-f655-416b-b1ae-4d27a4fe0f5a)


I made a quick reprex with the same 6x6 facet grid and some fake data:

```python
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

# Set the random seed for reproducibility
np.random.seed(42)

# Generate random data
n_points = 100
n_facets = 6 * 6

data = pd.DataFrame({
    'x': np.random.uniform(0, 2500, n_points * n_facets),
    'y': np.random.uniform(0, 2500, n_points * n_facets),
    'facet_col': np.tile(range(6), n_points * 6),
    'facet_row': np.repeat(range(6), n_points * 6)
})

# Set up the plot
sns.set(style="whitegrid")
g = sns.FacetGrid(data, col="facet_col", row="facet_row", height=3, aspect=1, 
                  sharex=False, sharey=False)

# Plot the data points
g.map(sns.scatterplot, "x", "y", alpha=0.6)

# Customize the plot
g.set_axis_labels("X", "Y")
g.set_titles(col_template="Col {col_name}", row_template="Row {row_name}")
g.fig.suptitle("6x6 Facetted Plot with Random Data", y=1.02)

# Adjust the layout and display the plot
plt.tight_layout()
plt.show()
```

Jupyter Notebook extension on left and Positron plot pane on right:

<img src="https://github.com/user-attachments/assets/c0fb442d-f635-4130-a2d2-70a4c73ea3bc" height=400 width=400>

<img src="https://github.com/user-attachments/assets/6cc80eed-6501-4438-a660-543048b2e4dd" height=400 width=400>

**Comment by tenFnAN** [comment]
Thank you very much for the tip. 
I felt a bit embarrassed for not solving the problem myself... I had the 'Auto' option selected... After switching to 'Matplotlib,' everything is fine. 
Thanks again!

**Reply by jthomasmock** [reply]
No embarrassment needed! The sizing dropdown is a brand-new concept if you're coming from the Notebook world. We think it's really valuable for customization, but if you really like the "default" behavior you can opt out of some of the more automatic or manual resizing we do on our side.
